pkg/loki/loki.go

```diff
@@ -481,7 +482,24 @@ func (t *Loki) setupModuleManager() error {
 		}
 	}
 
+	t.deps = deps
 	t.ModuleManager = mm
 
 	return nil
 }
+
+func (t *Loki) isModuleEnabled(m string) bool {
+	for _, tg := range t.Cfg.Target {
+		if tg == m {
+			return true
+		}
+		if k, ok := t.deps[tg]; ok {
+			for _, dp := range k {
+				if dp == m {
+					return true
+				}
+			}
+		}
+	}
+	return false
+}
```

Comment:
ohh, this is a much better check of dependencies, should we replace the function of the same name on the `Config` struct with this version and use that everywhere?

Reply:
Worried about introducing bugs as we close in on the 2.4 release so I created #4644 to track us doing this in the future.